src/track/trackref.cpp

```diff
@@ -0,0 +1,43 @@
+#include "track/trackref.h"
+
+
+bool TrackRef::verifyConsistency() const {
+    DEBUG_ASSERT_AND_HANDLE(hasLocation() || !hasCanonicalLocation()) {
+        return false;
+    }
+    DEBUG_ASSERT_AND_HANDLE(hasLocation() || !hasId()) {
+        return false;
+    }
+    return true;
+}
+
+QFileInfo TrackRef::createFileInfo() const {
+    QFileInfo fileInfo(getLocation());
```

Comment:
I have read a bit in the source and docs of QFileInfo and it turns out that it is very optimized. 
It does only things that are required to fulfill a query, but has a cache for results of system calls. 
The only mayor overhead in the constructor is checking for "\" or "/", and preparing the underlying FileEngine.

If we create temporary file Infos QFileInfo by calling createFileInfo() over and over again, we do not benefit from the caching nature of QFileInfo but we have always the penalty of constructing a new QFileInfo. 
However QFileInfo is not thread save. 

So we have on one other hand the locking inside Mixxx, to system calls, that are probably locking anyway, 
on the other hand we introduce CPU overhead by constructing QFileInfo. 
If we consider that the locks are never locked, because a TrackPointer is used only at on thread at a time, it is a performance regression. 

How about that: Add a QFileInfo member to TrackRef and reintroduce a mutex for it. 
Use it for all calls that require a system call, which is probably locking anyway.